### DISCUSSION TITLE
Bootstrap k3s with Embedded database other than etcd

### DISCUSSION THREAD

**Discussion by migs35323** [k3s-io/k3s#8745] [Q&A]
**Is your feature request related to a problem? Please describe.**
i have a lab composed of 2 machines (2 physical servers)
i intended it so in case one goes down i get the other as a backup.
i did setup k3s both as master servers with built in etcd however i noticed that in case 1 goes down, everything is down.
so i was trying to find alternative solutions to using etcd with it's requirement of odd nodes.
i know the best answer is "get a third machine", however i am requested to work in a 2 machine setup.
having just 1 as a master is not great because in case the master machine gets down, we can't easily join another machine to the cluster

it would be possible or feasable for k3s to bootstrap a embebbed sql (mariadb/posttgress/mysql) as the datastore?

**Describe the solution you'd like**

An option to bootstrap k3s with other database for example mariadb/postgresql/mysql

**Describe alternatives you've considered**

deploy the databases in the host machines and use them as endpoints or deploy them using docker 

I am thinking in deploying the database in docker however since k3s already has it's container management trough containerd, why not use it?

**Comment by brandond** [comment]
> having just 1 as a master is not great because in case the master machine gets down, we can't easily join another machine to the cluster

How does using external SQL DB make this any better? Even with an external DB, in a two-node cluster, there is still one machine (the database server) which must be up in order for the cluster to function.

If you want HA, you need at least 3 machines, two of which must be available.

HA with an SQL DB is designed for situations where you already have an externally-managed highly-available database platform available for use. If all you have is two nodes, putting the DB on one of them does not make things any better.

**Reply by migs35323** [reply]
thank you for your time to look at my request.
i was having the idea of both machines having an embebbed sql database like mariadb galera cluster (master-master), or a master-slave replication instead of the etcd cluster.
that's why i made the request, i would like to know if its possible to use a sql database running in both machines instead of a etcd cluster built in into the k3s, so if one machine goes away, the other one is still up and the cluster survives.
i understand the requirement of 3 machines for quorum, but it would be great if the cluster could survive with just 1 master(k3s server)
